src/PowerShellEditorServices.Hosting/SessionFileWriter.cs

```diff
@@ -0,0 +1,147 @@
+        /// <param name="debugAdapterTransport">The transport configuration for the debug adapter service.</param>
+        void WriteSessionStarted(ITransportConfig languageServiceTransport, ITransportConfig debugAdapterTransport);
+    }
+
+    /// <summary>
+    /// The default session file writer, which uses PowerShell to write a session file.
+    /// </summary>
+    public class SessionFileWriter : ISessionFileWriter
+    {
+        // Use BOM-less UTF-8 for session file
+        private static readonly Encoding s_sessionFileEncoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
+
+        private readonly HostLogger _logger;
+
+        private readonly string _sessionFilePath;
+
+        /// <summary>
+        /// Construct a new session file writer for the given session file path.
+        /// </summary>
+        /// <param name="logger">The logger to log actions with.</param>
+        /// <param name="sessionFilePath">The path to write the session file path to.</param>
+        public SessionFileWriter(HostLogger logger, string sessionFilePath)
+        {
+            _logger = logger;
+            _sessionFilePath = sessionFilePath;
+        }
+
+        /// <summary>
+        /// Write a startup failure to the session file.
+        /// </summary>
+        /// <param name="reason">The reason for the startup failure.</param>
+        /// <param name="details">Any extra details, which will be serialized as JSON.</param>
+        public void WriteSessionFailure(string reason, object details)
+        {
+            _logger.Log(PsesLogLevel.Diagnostic, "Writing session failure");
+
+            var sessionObject = new Dictionary<string, object>
+            {
+                { "status", "failed" },
+                { "reason", reason },
+            };
+
+            if (details != null)
+            {
+                sessionObject["details"] = details;
+            }
+
+            WriteSessionObject(sessionObject);
+        }
+
+        /// <summary>
+        /// Write a successful server startup to the session file.
+        /// </summary>
+        /// <param name="languageServiceTransport">The LSP service transport configuration.</param>
+        /// <param name="debugAdapterTransport">The debug adapter transport configuration.</param>
+        public void WriteSessionStarted(ITransportConfig languageServiceTransport, ITransportConfig debugAdapterTransport)
+        {
+            _logger.Log(PsesLogLevel.Diagnostic, "Writing session started");
+
+            var sessionObject = new Dictionary<string, object>
+            {
+                { "status", "started" },
+                { "languageServiceTransport", languageServiceTransport.SessionFileTransportName },
+                { "debugServiceTransport", debugAdapterTransport.SessionFileTransportName },
+            };
+
+            if (languageServiceTransport != null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this condition so that it does not always evaluate to 'true'.](https://app.codacy.com/app/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4579423)